tools/pepsi/src/gammaray.rs

```diff
@@ -0,0 +1,48 @@
+use std::{iter::repeat, path::PathBuf};
+
+use clap::Args;
+use color_eyre::{eyre::Context, Result};
+use constants::OS_VERSION;
+use futures::{stream::FuturesUnordered, StreamExt};
+use nao::Nao;
+use repository::get_image_path;
+
+use crate::{parsers::NaoAddress, results::gather_results};
+
+#[derive(Args)]
+pub struct Arguments {
+    /// Alternative path to an image
+    #[arg(long)]
+    image_path: Option<PathBuf>,
+    /// Alternative HULKs-OS version e.g. 3.3
+    #[arg(long)]
+    os_version: Option<String>,
+    /// The NAOs to flash the image to, e.g. 20w or 10.1.24.22
+    #[arg(required = true)]
+    naos: Vec<NaoAddress>,
+}
+
+pub async fn gammaray(arguments: Arguments) -> Result<()> {
+    let version = arguments.os_version.as_deref().unwrap_or(OS_VERSION);
+    let image_path = match arguments.image_path {
+        Some(image_path) => image_path,
+        None => get_image_path(version).await?,
+    };
```

Comment:
I don't want `get_image_path()` to be executed if an image path is given, as it may download an image if not already done. So `unwrap_or_else()` would usually be my choice for that, but I don't know whether this is possible in this case, since `get_image_path()` is async and async closures currently unstable.

Reply:
Something like this might be possible. But I agree, that this also is not an improvement.
```rust
    let image_path = ready(arguments.image_path.ok_or(eyre!("")))
        .or_else(|_| get_image_path(version))
        .await?;
```
Feel free to resolve this discussion, if read